### PR TITLE
Update Plugins for QGIS 3 in PyQGIS Cookbook

### PR DESCRIPTION
**Pull request by cjmayo**
- Be clearer in the examples about which files are mandatory.

- Mention the Log Messages Plugins panel.

- Remove references to the Python Plugin Repositories wiki.

- Use https where available.

- Minor wording improvements.

### CODE REVIEW

**Comment by DelazJ** [comment]
you should also update the page status warning

File: source/docs/pyqgis_developer_cookbook/plugins.rst [hunk 1]
```diff
@@ -40,11 +40,11 @@ Steps:
...
    Is there already another plugin for that problem?
-#. *Create files*: Create the files described next.
-   A starting point (:file:`__init__.py`).
-   Fill in the :ref:`plugin_metadata` (:file:`metadata.txt`)
-   A main python plugin body (:file:`mainplugin.py`).
-   A form in QT-Designer (:file:`form.ui`), with its :file:`resources.qrc`.
+#. *Create files*: The essentials:
+   a starting point (:file:`__init__.py`);
+   fill in the :ref:`plugin_metadata` (:file:`metadata.txt`).
+   Then your own design. A main Python plugin body (e.g. :file:`mainplugin.py`).
```
[the remark this answers is not in the document]
Comment (cjmayo) [reply source/docs/pyqgis_developer_cookbook/plugins.rst]: Added implement. Also removed the brackets around the filenames, they are not even used throughout this section.

File: source/docs/pyqgis_developer_cookbook/plugins.rst [hunk 2]
```diff
@@ -103,9 +99,10 @@ What is the meaning of the files:
 is an online automated way of creating the basic files (skeleton) of a typical
 QGIS Python plugin.
 
-Also there is a QGIS plugin called `Plugin Builder <http://geoapt.net/pluginbuilder/>`_
-that creates plugin template from QGIS and doesn't require internet connection.
-This is the recommended option, as it produces 2.0 compatible sources.
+There is a QGIS plugin called
+`Plugin Builder 3 <https://plugins.qgis.org/plugins/pluginbuilder3/>`_
+that creates a plugin template for QGIS and doesn't require an internet connection.
+This is the recommended option, as it produces 3.0 compatible sources.
```
[the remark this answers is not in the document]
Comment (cjmayo) [reply source/docs/pyqgis_developer_cookbook/plugins.rst]: Yes, changed.

**Comment by cjmayo** [comment]
Status warning removed.

**Comment by DelazJ** [comment]
Thanks!

**Pull request merged**